test/libsolidity/ABIJson/function_type_extended.sol

```diff
@@ -0,0 +1,23 @@
+contract test {
+    function g(function(test) external returns (test[] memory) x) public {}
+}
+// ----
+//     :test
+// [
+//   {
+//     "constant": false,
+//     "inputs":
+//     [
+//       {
+//         "internalType": "function (contract test) external returns (contract test[])",
```

Comment:
Now that I'm reading it: maybe prefix with `<type>_` instead, such that it follows the compiler-internal rules for identifiers (i.e. `function (contract_test) external returns (struct_C.T[])`)? This could make a difference for people when they have to write parsers for it.

But I'm also fine with the above :)

Reply:
This is just the `toString` method of types - people might already have parsers for that because this is also what is used in the AST, I think.